dhcp/pxe.go

```diff
@@ -123,14 +134,19 @@ func SetupPXE(ctx context.Context, rep, req *dhcp4.Packet) bool {
 	return true
 }
 
-func SetFilename(rep *dhcp4.Packet, filename string, nextServer net.IP, pxeClient bool) {
+func SetFilename(rep *dhcp4.Packet, filename string, nextServer net.IP, httpServerFQDN string, httpClient bool) {
+	if httpClient {
+		filename = "http://" + httpServerFQDN + "/" + filename
+	}
 	file := rep.File()
 	if len(filename) > len(file) {
 		err := errors.New("filename too long, would be truncated")
 		// req CHaddr and XID == req's
 		dhcplog.With("mac", rep.GetCHAddr(), "xid", rep.GetXID(), "filename", filename).Fatal(err)
 	}
-	if pxeClient {
+	if httpClient {
```

Comment:
This was never moved up right? I'm on the fence about setting things over 2 different places vs not. Also, I think the `if pxeClient` check should stick around. Maybe we can have the function look like this:

```go

func SetFilename(rep *dhcp4.Packet, filename string, nextServer net.IP, httpServerFQDN string, pxeClient, httpClient bool) {
	rep.SetSIAddr(nextServer) // next-server: IP address of the TFTP/HTTP Server.

	var optionClassID string
	switch {
	case httpClient:
		filename = "http://" + httpServerFQDN + "/" + filename
		optionClassID = "HTTPClient"
	case pxeClient:
		optionClassID = "PXEClient"
	}
	if optionClassID != "" {
		rep.SetString(dhcp4.OptionClassID, optionClassID)
	}

	file := rep.File()
	if len(filename) > len(file) {
		err := errors.New("filename too long, would be truncated")
		// req CHaddr and XID == req's
		dhcplog.With("mac", rep.GetCHAddr(), "xid", rep.GetXID(), "filename", filename).Fatal(err)
	}
	copy(file, filename) // filename: Executable (or iPXE script) to boot from.
}
```
This way we're not mixing filename override & len checks with optionclassid stuff, etc. This now keeps it modification of the reply packet a little apart and is easier to read I think. It also brings back the `pxeClient` bool because I'm not sure its ok to drop it.

Reply:
with a log message as the default case too actually.

Reply:
I missed this. Its now changed as requested, but I'm still keeping the `httpClient` bool as I think having two booleans is misleading. What might be done instead if having an `clientType` kinda of enum. But I'm not doing that right now.

Reply:
Looking through it I think this is fine `dhcp.SetFilename` is only called from `job.setPXEFilename` which is called from `job.configureDHCP` but only after verifying that its a PXE request. So until another transfer method is used defaulting to tftp/PXEClient is fine.